### PR TITLE
feat: add or amend claim documents event (CMC-1198)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-1198

### Change description ###

New add or amend claim documents event in AWAITING CASE NOTIFICATION state
Updated servedDocumentFiles allowed files
Updated particulars of claim field from collection to single upload and typed entry
Validation that either document or typed entry is added

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/api/steps.js [hunk 1]
```diff
@@ -72,6 +73,29 @@ module.exports = {
...
 
+  amendClaimDocuments: async () => {
+    eventName = 'ADD_OR_AMEND_CLAIM_DOCUMENTS';
+    let returnedCaseData = await apiRequest.startEvent(eventName, caseId);
+    assertContainsPopulatedFields(returnedCaseData);
+    caseData = returnedCaseData;
+
+    await validateEventPages(data[eventName]);
+
+    await assertCallbackError('Upload', data[eventName].invalid.Upload.duplicateError,
```
Comment (ckalista) [inline e2e/api/steps.js:92]: 👍

File: e2e/api/steps.js [hunk 2]
```diff
@@ -72,6 +73,29 @@ module.exports = {
...
+    await assertCallbackError('Upload', data[eventName].invalid.Upload.nullError,
+      'One particular of claim is required');
+
+    await assertSubmittedEvent('AWAITING_CASE_NOTIFICATION', {
+      header: 'Documents uploaded successfully',
+      body: '<br />'
+    }, true);
+
+    await assertCorrectEventsAreAvailableToUser(config.solicitorUser, 'AWAITING_CASE_NOTIFICATION');
+    await assertCorrectEventsAreAvailableToUser(config.defendantSolicitorUser, 'AWAITING_CASE_NOTIFICATION');
```
Comment (ckalista) [inline e2e/api/steps.js:104]: Aren't we missing a few here: 
image

Comment (ckalista) [reply e2e/api/steps.js:104]: Same are missing in `createClaimWithRepresentedRespondent`.

**Review by ckalista — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/model/ServedDocumentFiles.java [hunk 3]
```diff
@@ -5,15 +5,31 @@
...
 
     private List<Element<Document>> other;
     private List<Element<Document>> medicalReports;
     private List<Element<Document>> scheduleOfLoss;
-    private List<Element<Document>> particularsOfClaim;
+    private Document particularsOfClaimDocument;
+    private String particularsOfClaimText;
     private List<Element<Document>> certificateOfSuitability;
+
+    public List<String> getErrors() {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/model/ServedDocumentFiles.java:26]: should we add `@JsonIgnore` on this?

**Review by sabahirfan — DISMISSED** [review]

File: e2e/api/steps.js (hunk 2 above)
Comment (swalker125) [reply e2e/api/steps.js:104]: the assertion asserts for state, so we are asserting all these events are available 
```  
AWAITING_CASE_NOTIFICATION: [
    events.NOTIFY_DEFENDANT_OF_CLAIM,
    events.ADD_DEFENDANT_LITIGATION_FRIEND,
    events.CASE_PROCEEDS_IN_CASEMAN,
    events.ADD_OR_AMEND_CLAIM_DOCUMENTS
  ],
```

**Review by douglasrice — APPROVED** [review]

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**